hack/tpm2/main.go

```diff
@@ -0,0 +1,809 @@
+package main
+
+// TODO: maybe some of this is useful: https://ericchiang.github.io/post/tpm-keys/
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/x509"
+	_ "embed"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"log"
+
+	"github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpm2/transport"
+	"github.com/google/go-tpm/tpm2/transport/simulator"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+var (
+	//go:embed testdata/tpm2-pcr-public.pem
+	pcrSigningPubKey []byte
+	//go:embed testdata/pcr-data.json
+	pcrDataJSON []byte
+)
+
+type LuksHeader struct {
+	Type            string   `json:"type"`
+	Keyslots        []string `json:"keyslots"`
+	TPM2PrivateBlob []byte   `json:"tpm2_private_blob"`
+	TPM2PublicBlob  []byte   `json:"tpm2_public_blob"`
+	TPM2PCRS        []int    `json:"tpm2_pcrs"`
+	TPM2Alg         string   `json:"tpm2_alg"`
+	TPM2PolicyHash  []byte   `json:"tpm2_policy_hash"`
+	TPM2PublicKey   string   `json:"tpm2_public_key"`
```

Comment:
this is not used, not sure if we need to save that in luks header, the only thing we can do is validate the public key did not change for a new uki, but that would prevent from using a new signing key at all. So I guess we can drop it.